### PR TITLE
Date range picker component

### PR DESCRIPTION
**Pull request by zagovorichev**
https://oat-sa.atlassian.net/browse/TAO-2846

### CODE REVIEW

File: views/js/test/ui/dateRange/test.html [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <meta charset="utf-8">
+    <title>Ui Test - Deleter UI Component</title>
```
Comment (krampstudio) [inline views/js/test/ui/dateRange/test.html]: The page title is incorrect

File: views/js/test/ui/dateRange/test.html [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    <base href="../../../../"/>
+    <link rel="stylesheet" type="text/css" href="js/lib/qunit/qunit.css">
+    <link rel="stylesheet" type="text/css" href="css/tao-main-style.css">
+    <script type="text/javascript" src="js/lib/require.js"></script>
+    <script type="text/javascript" src="js/lib/qunit/qunit.js"></script>
```
Comment (krampstudio) [inline views/js/test/ui/dateRange/test.html]: could you also add the coverage's script

File: views/js/ui/dateRange.js [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+            }
+        };
+
+        var hasInputs = function hasInputs() {
+            if (initConfig.startInput || initConfig.endInput) {
+                if (initConfig.startInput && initConfig.endInput) {
+                    return true;
+                } else {
+                    if (window.console) {
+                        console.log('You should determine both of the startInput and endInput as a configuration options');
```
Comment (krampstudio) [inline views/js/ui/dateRange.js]: I'm not sure to understand whom this message is supposed to target ? the implementer, the user ?

File: views/js/ui/dateRange.js [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+                    }
+                }
+            }
+            return false;
+        };
+
+        $componentDateRange = component(dateRange);
```
Comment (krampstudio) [inline views/js/ui/dateRange.js]: variables starting by `$` are reserved by convention to jquery elements

File: views/js/ui/dateRange.js [hunk 5]
```diff
@@ -0,0 +1,329 @@
...
+             * Calls `method` on the `startTime` and `endTime` elements, and configures them to
+             * enforce date range limits.
+             * @param  string method Can be used to specify the type of picker to be added
+             * @param  Element startTime
+             * @param  Element endTime
+             * @param  obj options Options for the `timepicker()` call. Also supports `reformat`,
+             *   a boolean value that can be used to reformat the input values to the `dateFormat`.
+             * @return jQuery
+             */
+            $.timepicker.triggeredHandleRange = function (method, startTime, endTime, options) {
```
Comment (krampstudio) [inline views/js/ui/dateRange.js]: please name lambda functions : 
```js
$.timepicker.triggeredHandleRange = function triggeredHandleRange(method, startTime, endTime, options)
```

File: views/js/ui/dateRange.js [hunk 6]
```diff
@@ -0,0 +1,329 @@
...
+            /**
+             * Gets the end date of the range
+             * @returns {String}
+             */
+            getEnd: function getEnd() {
+                return periodEnd;
+            }
+        };
+
+        var hasInputs = function hasInputs() {
```
Comment (krampstudio) [inline views/js/ui/dateRange.js]: missing jsdoc

File: views/js/ui/dateRange.js [hunk 7]
```diff
@@ -0,0 +1,329 @@
...
+
+                        selected($(this), startTime, 'maxDate', false);
+                    }
+                }, options, options.end));
+
+                checkDates(startTime, endTime);
+                selected(startTime, endTime, 'minDate', true);
+                selected(endTime, startTime, 'maxDate', false);
+
+                function checkDates(changed, other) {
```
Comment (krampstudio) [inline views/js/ui/dateRange.js]: missing jsdoc

File: views/js/ui/dateRange.js [hunk 8]
```diff
@@ -0,0 +1,329 @@
...
+                        else if (options.maxInterval > 0 && maxDate < enddt) { // max interval check
+                            endTime[method]('setDate', maxDate);
+                        }
+                        else if (startdt > enddt) {
+                            other[method]('setDate', changeddt);
+                        }
+                    }
+                }
+
+                function selected(changed, other, option, isStart) {
```
Comment (krampstudio) [inline views/js/ui/dateRange.js]: missing jsdoc

File: views/js/ui/dateRange.js [hunk 9]
```diff
@@ -0,0 +1,329 @@
...
+                    if (!changed.val()) {
+                        return;
+                    }
+                    var date = changed[method].call(changed, 'getDate');
```
Comment (krampstudio) [inline views/js/ui/dateRange.js]: please declare variables on top of their scope to reflect hoisting

File: views/js/ui/dateRange.js (hunk 3 above)
Comment (zagovorichev) [inline views/js/ui/dateRange.js]: For implementer if he determine only one input

**Comment by zagovorichev** [comment]
@krampstudio Update

File: views/js/ui/dateRange.js (hunk 3 above)
Comment (krampstudio) [inline views/js/ui/dateRange.js]: Then throw a `TypeError`

Comment (zagovorichev) [inline views/js/ui/dateRange.js]: But function return bool, I don't want catch errors

Comment (krampstudio) [inline views/js/ui/dateRange.js]: Then do not use console.log

**Comment by zagovorichev** [comment]
@krampstudio Update

**Pull request merged**